### ISSUE TITLE
CAIRO_STATUS_NO_MEMORY when using @font-face

### ISSUE THREAD

**Issue by cuu508** [Kozea/WeasyPrint#441]
weasyprint sometimes crashes with a "CAIRO_STATUS_NO_MEMORY" error when rendering documents with @font-face CSS rules.

The crashes are not deterministic, same document may render fine 100 times, and crash on the 101th time.

Test case:

```
from weasyprint import HTML

html = b"""
<style>
  @font-face {
    font-family: DejaVu;
    src: local(DejaVu Sans);
  }
  html {
    font-family: DejaVu;
  }
</style>
lorem ipsum
<img src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+Cjxzdmcgd2lkdGg9IjEiIGhlaWdodD0iMSI+PHRleHQ+YTwvdGV4dD48L3N2Zz4K"/>
"""

for i in range(0, 1000):
    print("Iteration %d" % i)
    HTML(string=html).write_pdf()
```

Some previous discussion is in #103

**Comment by liZe** [comment]
Various hints for anyone interested in fixing this issue:
- Only happens with both HTML text using `@font-face` and SVG rendering with text (not related to the font family used in the SVG).
- Only happens when the SVG is rendered after the HTML text (only a Pango warning otherwise).
- Only happens when writing a PDF (not with PNG, not with only `Document.render()`).

**Comment by Mihara** [comment]
Ow. I just stepped on that.
Somehow, I wasn't able to stop it from crashing even when excising all SVG from output. Only switching to loading fonts with `local()` prevented crashes reliably.

**Comment by liZe** [comment]
That's magically fixed by f5367c8.

Other rendering errors could happen before, but they were not fatal as this one. The font was sometimes removed from filesystem before the surface was written on the output PDF/image.
